File: js/uniqSubs.js
function uniqSubs(str){
  const uniqs = new Set();

  for (let i = 0; i < str.length; i++) {
    for (let j = i + 1; j <= str.length; j++) {
      uniqs.add(str.slice(i, j));
    }
  }

  return uniqs;
}


console.log(uniqSubs("bobobobob"));

File: ruby/uniq_subs.rb
require 'set'

def uniq_subs(str)
  uniqs = Set.new()

  (0...str.length).each do |i|
    (i + 1..str.length).each do |j|
      uniqs << str[i...j]
    end
  end

  uniqs
end

p uniq_subs("bobobobob")
